tester_script.py

```diff
@@ -0,0 +1,219 @@
+    try:
+        if not isinstance(ds.scene_mapping, dict):
+            print('--- Scene_mapping must be type: dict ---')
+            print('ERROR: Currently of type:', type(ds.scene_mapping), '\n')
+    except AttributeError:
+       pass 
+
+
+    # testing supercategories_to_names
+    try:
+        if not isinstance(ds.supercategories_to_names, dict): 
+            print('ERROR: self.supercategories_to_names must be type: dict \n')
+    except AttributeError:
+        print('ERROR: self.supercategories_to_names is a required field.')
+        print('Default to DEFAULT_GROUPINGS_TO_NAMES\n')
+
+    
+    # testing __len__:
+    try: 
+        ds.__len__()
+        if ds.__len__() != len(ds.image_ids):
+            print('ERROR: self.__len__() must be equal to length of self.image_ids')
+            print('self.__len__() returns', ds.__len__(), '/ length of self.image_ids =', len(ds.image_ids), '\n')
+    except AttributeError: 
+        print('ERROR: self.__len__() is a required method.\n')
+ 
+
+    # testing __getitem__ and from_path
+    rand_ind = random.randint(0, ds.__len__()-1)
+    try: 
+        x = ds.__getitem__(rand_ind)
+    except AttributeError: 
+        print('ERROR: self.__getitem___() is a required method.\n')
+   
+    x = ds.__getitem__(rand_ind)
+    if len(x) != 2: 
+        print('ERROR: self.__getitem__() must return a tuple of length: 2')
+        print('Return value should be in form (image, annotations)\n')
+        sys.exit()
+    img, anns = x
+    if not isinstance(anns, list): 
+        print('ERROR: Annotations must be of type: list\n') 
+        sys.exit()
+    if len(anns) != 5:
+        print('ERROR: self.__getitem__() should return annotations of length: 5')
+        print('Annotations must be a list containing [image_anns, gender_info, [country, lat_lng], file_path, scene_group]\n')
+        sys.exit()
+    labels, att, geo, fp, scn = anns
+    if len(labels) > 1: 
+        if not isinstance(labels[0], dict) or labels[0].get('label', None)==None: 
+            print('ERROR: image_anns must be a list of dicts. If there are >0 dicts, must contain keyword \'label\' \n')
+    for label in labels: 
+        wrong_bbox=None
+        if label.get('bbox', None): 
+            for coord in label['bbox']: wrong_bbox=label['bbox'] if (coord<0 or coord>1) else None
+    if wrong_bbox: 
+        print('ERROR: All bounding box numbers must be scaled between 0 and 1. Got bounding box: ', end='')
+        for coord in wrong_bbox: 
+            print('%.2f, ' % coord, end='')
+        print('\n')
+    if (len(att) != 1 and len(att) != 2) or (not isinstance(att[0], int)): 
+        print('ERROR: gender_info must be a list of length: 1 in the form [None] or [int] or length: 2 in the form [int, [double, double, double, double]]\n')
+        print('Got:', att[0]) 
+    if att[0] != 0 and att[0]!= 1: 
+        print('ERROR: gender annotations should be 0 for male, 1 for female, or None. Got value:', att[0], '\n')
```

Comment:
should call self.attribute_names or something like that to get the names of attributes